tests/src/performance/JitBench/JitBenchHarness.cs

```diff
@@ -0,0 +1,355 @@
+        }
+
+        private static ProcessStartInfo Setup()
+        {
+            PrintHeader("Starting SETUP");
+
+            var dotnetDirectory = Path.Combine(s_jitBenchDevDirectory, ".dotnet");
+            var dotnetProcessFileName = Path.Combine(dotnetDirectory, "dotnet.exe");
+
+            //Download and extract the repo.
+            using (var client = new HttpClient())
+            {
+                // Download the JitBench repository and extract it.
+                var url = @"https://github.com/guhuro/JitBench/archive/dev.zip"; // TODO: This should be updated to https://github.com/aspnet/JitBench/archive/dev.zip once the fork is merged.
+                var zipFile = Path.Combine(s_temporaryDirectory, "dev.zip");
+
+                using (var tmpzip = File.Create(zipFile))
+                {
+                    using (var stream = client.GetStreamAsync(url).Result)
+                        stream.CopyTo(tmpzip);
+                    tmpzip.Flush();
+                }
+
+                // If the repo already exists, we delete it and extract it again.
+                if (Directory.Exists(s_jitBenchDevDirectory))
+                    Directory.Delete(s_jitBenchDevDirectory, true);
+
+                // This step will create s_JitBenchDevDirectory.
+                ZipFile.ExtractToDirectory(zipFile, s_temporaryDirectory);
+            }
+
+            // first dotnet-install.ps1
+            var installProcessStartInfo = new ProcessStartInfo() {
+                WorkingDirectory = s_jitBenchDevDirectory,
+                FileName = @"powershell.exe",
+                Arguments = @".\Dotnet-Install.ps1 -SharedRuntime -InstallDir .dotnet -Channel master -Architecture x64"
+            };
+            LaunchProcess(installProcessStartInfo, 180000);
+
+            // second dotnet-install.ps1
+            var secondInstallProcessStartInfo = new ProcessStartInfo() {
+                WorkingDirectory = s_jitBenchDevDirectory,
+                FileName = @"powershell.exe",
+                Arguments = @".\Dotnet-Install.ps1 -InstallDir .dotnet -Channel master -Architecture x64"
+            };
+            LaunchProcess(secondInstallProcessStartInfo, 180000);
+
+            // dotnet restore
+            var musicStoreDirectory = Path.Combine(s_jitBenchDevDirectory, "src", "MusicStore");
+            var restoreProcessStartInfo = new ProcessStartInfo() {
+                WorkingDirectory = musicStoreDirectory,
+                FileName = dotnetProcessFileName,
+                Arguments = "restore"
+            };
+            LaunchProcess(restoreProcessStartInfo, 300000);
+
+            // Modifying the shared framework
+            var sourcePath = Directory.GetCurrentDirectory();
+            var targetPath = Path.Combine(s_jitBenchDevDirectory, ".dotnet", "shared", "Microsoft.NETCore.App");
+            var di = new DirectoryInfo(targetPath);
+            targetPath = Path.Combine(targetPath, di.GetDirectories("2.0*")[0].Name);
```

Comment:
This is somewhat dangerous; it assumes that MusicStore runs using the 2.0 shared fx. Could we somehow assert the condition? Or, if we have to do our best with a default, could we just take the latest (instead of the first/oldest 2.0) version?